src/ol/source/Google.js

```diff
@@ -238,10 +252,48 @@ class Google extends TileImage {
     const timeout = Math.max(expiry - Date.now() - 60 * 1000, 1);
     this.sessionRefreshId_ = setTimeout(() => this.createSession_(), timeout);
 
+    this.setAttributions(this.fetchAttributions.bind(this));
     // even if the state is already ready, we want the change event
     this.setState('ready');
   }
 
+  async fetchAttributions(frameState) {
+    if (
+      frameState.viewHints[ViewHint.ANIMATING] ||
+      frameState.viewHints[ViewHint.INTERACTING]
+    ) {
+      return this.previousViewportAttribution_;
+    }
+    const transform = getTransformFromProjections(
+      this.getProjection(),
+      getProjection('EPSG:4326'),
+    );
+    const epsg4326Extent = applyTransform(frameState.extent, transform);
+    const tileGrid = this.getTileGrid();
+    const zoom = tileGrid.getZForResolution(
+      frameState.viewState.resolution,
+      this.zDirection,
+    );
+    const north = epsg4326Extent[3];
+    const south = epsg4326Extent[1];
+    const east = epsg4326Extent[2];
+    const west = epsg4326Extent[0];
+    const viewportExtent = `zoom=${zoom}&north=${north}&south=${south}&east=${east}&west=${west}`;
+    // check if the extent or zoom has actually changed to avoid unnecessary requests
+    if (this.previousViewportExtent_ == viewportExtent) {
+      return this.previousViewportAttribution_;
+    }
+    this.previousViewportExtent_ = viewportExtent;
```

Comment:
Not needed if the above works.

Reply:
Tried it, it removed most of the duplicates (like I don't get duplicates when panning the map), but it gets me one duplicate when zooming (I guess there is an event when zoom starts and when it ends), so I still need a check to abort the unnecessary request. I added the `frameState.animate` check for early termination but kept the this.previousViewportExtent check that was there